lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2758,11 +2784,28 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
                     db currentEpochSlotting knownPools
                     poolStatus withdrawal
 
+        optionalVoteAction <- case (body ^. #vote) of
+            Just (ApiT action) ->
+                liftIO $ Just <$> WD.voteAction trWorker db action
+            Nothing -> do
+                areWeInConway <- liftIO $ W.isAlreadyConwayEra nl
+                haveWeVoted <- liftIO $ W.haveWeVoted db
+                stakingKeyRegistered <- liftIO $ W.isStakeKeyInDb db
+                if (areWeInConway && not haveWeVoted && isJust (body ^. #delegations)) then
+                   if stakingKeyRegistered then
+                       pure $ Just $ Vote Abstain
+                   else
+                       pure $ Just $ VoteRegisteringKey Abstain
+                else
+                   pure Nothing
```

Comment:
Please move this functionality further down into `Cardano.Wallet`. In addition, please make this logic part of a **pure** function.

I think that the right place to put this is the new function [`certificateFromDelegationAction`][1], as this function is dedicated to the following problem "Given that the user wants to perform `DelegationAction`, what kind of certificate do we need to put into the transaction?". In the Conway era, this function will add the vote to abstain.

  [1]: https://github.com/cardano-foundation/cardano-wallet/blob/038b952948cd53fb7ff5548b8a3110a874d870d3/lib/wallet/src/Cardano/Wallet/Transaction/Delegation.hs#L51